docs/manual/codespaces.md

```diff
@@ -0,0 +1,273 @@
+<small>Machine type: 2 vCPU · 8 GB RAM</small>
+</div>
+///
+/// tab | Button code
+**Do not forget to change the lab repo URL and machine type in the code below!**
+
+```html
+---
+<div align=center markdown>
+<a href="https://codespaces.new/srl-labs/srlinux-vlan-handling-lab?quickstart=1">
+<img src="https://gitlab.com/rdodin/pics/-/wikis/uploads/d78a6f9f6869b3ac3c286928dd52fa08/run_in_codespaces-v1.svg?sanitize=true" style="width:50%"/></a>
+
+**[Run](https://codespaces.new/srl-labs/srlinux-vlan-handling-lab?quickstart=1) this lab in GitHub Codespaces for free**.  
+[Learn more](https://containerlab.dev/manual/codespaces) about Containerlab for Codespaces.  
+<small>Machine type: 2 vCPU · 8 GB RAM</small>
+</div>
+
+---
+```
+
+///
+
+Check out [srl-labs/srl-streaming-telemetry](https://github.com/srl-labs/srl-telemetry-lab?tab=readme-ov-file#nokia-sr-linux-streaming-telemetry-lab) README where this button is used to start the lab in Codespaces.
+
+And of course, you can always launch the Codespace using the GitHub UI by clicking on the "Code" button.
+
+![start](https://gitlab.com/rdodin/pics/-/wikis/uploads/34a9e417d61c02cf9a4283cc1c2cf8cb/image.png)
+
+1. Click on the "Code" button in the GitHub UI.
+2. Start the Codespace on the `main` branch using the devcontainer settings defined in the `.devcontainer/devcontainer.json` file.
+3. Or open up an advanced menu
+4. And configure the Codespace settings manually.
+
+## Dev Container
+
+The key pillar behind Codespaces is the Containerlab' Dev Container image that defines the environment in which the lab will run. The Dev Container image is a Docker image that contains all the necessary tools and dependencies to run Containerlab and other tools that you might need in the lab.
+
+The following two files define Containerlab's Dev Container image:
+
+1. [devcontainer.json](https://github.com/srl-labs/containerlab/blob/main/.devcontainer/devcontainer.json) - the Dev Container configuration file that defines how the environment is built, configured and launched.
+2. [Dockerfile](https://github.com/srl-labs/containerlab/blob/main/.devcontainer/Dockerfile) - the Dockerfile that the Dev Container is built from.
+
+The resulting Dev Container image contains the following tools and dependencies:
+
+- containerlab binary installed via the deb repository
+- docker in docker setup
+- [gNMIc](https://gnmic.openconfig.net) and [gNOIc](https://gnoic.kmrd.dev) tools
+- Go SDK
+- Python 3 with pyenv
+- `gh` CLI tool
+- zsh shell with oh-my-zsh configuration
+- VS Code plugins
+
+## Tips, tricks and known issues
+
+### Authenticating with ghcr.io container registry
+
+While we still live in a world where vendor' Network OS images are not available in the public container registries, you might want to push the images you've built to your GitHub container registry as a `private` image. That way you get to benefit from building the image once and reusing it across different labs while keeping the image private.
```

Comment:
NO-NO-NO! Please remove that before some legal department saw this. =)
The fact that some images are not on container registries is fundamentally wrong, but we can not advise that to users. Even though this is very theoretical - they are legally liable if they upload images anywhere as normally it's covered by EULA agreements etc.
We can however advice upload directly to a codespace and I can add a section about using [EOS downloader](https://pypi.org/project/eos-downloader/) later.

Reply:
Maybe also add a section where other people/vendors can index their repositories with Codespaces support and provide corresponding instructions. I'll be very happy to list mine.
If there is nothing listed - the community can poke the vendor with a stick if there will be demand.